app/controllers/IncomeController.scala

```diff
@@ -93,7 +93,7 @@ class IncomeController @Inject()(personService: PersonService,
           for {
             cachedData <- journeyCacheService.mandatoryValues(UpdateIncome_NameKey, UpdateIncome_IdKey, UpdateIncome_ConfirmedNewAmountKey)
           } yield {
-            val model = SameEstimatedPayViewModel(cachedData(0), cachedData(1).toInt, cachedData(2).toInt, false)
+            val model = SameEstimatedPayViewModel(cachedData(0), cachedData(1).toInt, cachedData(2).replaceAll("""\D""", "").toInt, false)
             Ok(views.html.incomes.sameEstimatedPay(model))
```

Comment:
Couple of things here: 

You should extract `cachedData(2).replaceAll("""\D""", "")` here to a descriptive variable. You can also use FormHelper.stripNumber(cachedData(2)) to get a machine friendly number. 

This isn't the right place to "clean" the data, it should be at the point where UpdateIncome_ConfirmedNewAmountKey is cached. Otherwise where it's used everywhere else might cause the same issue.